modules/ocf_filehost/manifests/init.pp

```diff
@@ -1,11 +1,43 @@
-class ocf_filehost {
+class ocf_filehost(
+  String $storage_device
+) {
   package { ['quotatool', 'nfs-kernel-server']:; }
 
-  file { '/etc/exports':
-    source  => 'puppet:///modules/ocf_filehost/exports',
-    require => Package['nfs-kernel-server'],
+  file {
+    '/etc/exports':
+      source  => 'puppet:///modules/ocf_filehost/exports',
+      require => Package['nfs-kernel-server'];
+
+    '/opt/homes':
+      ensure  => directory;
   }
 
+  mount {
+    '/opt/homes':
+      ensure  => mounted,
+      atboot  => true,
+      device  => $storage_device,
+      fstype  => 'ext4',
+      options => 'noacl,noatime,nodev,usrquota';
```

Comment:
If we need to mount NFS read only, will this automatically change mount options?

Reply:
we'd probably mount NFS ro on the clients, right? not on the host?

Reply:
Looks like in any case puppet won't remount ro -> rw, so this point is moot.